### PR TITLE
pkgsStatic.toybox: fix build

### PR DESCRIPTION
**Pull request by alyssais**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/toybox/default.nix [hunk 1]
```diff
@@ -18,7 +18,8 @@ stdenv.mkDerivation rec {
   };
 
   depsBuildBuild = [ buildPackages.stdenv.cc ]; # needed for cross
-  buildInputs = lib.optionals enableStatic [ stdenv.cc.libc stdenv.cc.libc.static ];
+  buildInputs = lib.optionals (enableStatic && stdenv.cc.libc ? static)
+    [ stdenv.cc.libc stdenv.cc.libc.static ];
```
Comment (r-burns) [inline pkgs/tools/misc/toybox/default.nix:22]:
```suggestion
```
I think we can just remove this entirely? Seems weird to specify libc at all, it should just be part of cc afaik.

Comment (alyssais) [reply pkgs/tools/misc/toybox/default.nix:22]: Nope, removing either element breaks static glibc builds with `toybox.override { enableStatic = true; }`.

Comment (r-burns) [reply pkgs/tools/misc/toybox/default.nix:22]: Ah I see, thank you

**Review by r-burns — APPROVED** [review]
Tested on NixOS x86_64

**Pull request merged**